### PR TITLE
Doctor should check for Liquid errors

### PR DESCRIPTION
**Pull request by pathawks**
`jekyll doctor` Should try to generate the site using Liquid `:strict` mode and report any errors.

#4372

### CODE REVIEW

**Comment by pathawks** [comment]
Unfortunately, it's not working. This isn't going to be as simple as I had expected. :rage1:

File: test/test_doctor_command.rb [hunk 1]
```diff
@@ -33,4 +33,22 @@ class TestDoctorCommand < JekyllUnitTest
...
+    should 'return error for invalid Liquid' do
+      @site = Site.new(Jekyll.configuration({
+        "source" => File.join(source_dir, '/_liquid_errors'),
+        "destination" => dest_dir
+      }))
+      @site.process
+      output = capture_stderr do
+         ret = Jekyll::Commands::Doctor.liquid_errors(@site)
+         assert_equal true, ret
+      end
```
Comment (parkr) [inline test/test_doctor_command.rb]: Could you assert a particular output? We want to make sure these errors are surfaced.

**Comment by parkr** [comment]
:tada:

File: test/test_doctor_command.rb [hunk 2]
```diff
@@ -33,4 +33,22 @@ class TestDoctorCommand < JekyllUnitTest
...
   end
+
+  context 'Liquid errors' do
+    setup do
+      clear_dest
+    end
+
+    should 'return error for invalid Liquid' do
+      @site = Site.new(Jekyll.configuration({
+        "source" => File.join(source_dir, '/_liquid_errors'),
```
Comment (parkr) [inline test/test_doctor_command.rb]: Does `test/source/_liquid_errors` exist?

Comment (pathawks) [inline test/test_doctor_command.rb]: Added in this PR, yes.

File: test/test_doctor_command.rb (hunk 1 above)
Comment (pathawks) [inline test/test_doctor_command.rb]: I *could*, but right now there is no output. It is not throwing any errors or anything.

Comment (parkr) [inline test/test_doctor_command.rb]: That's not good! If there's no output, then the user has no idea why the build failed.

**Comment by pathawks** [comment]
I'm in too deep here. :sweat:

**Pull request closed without merging**